src/main/java/com/puppycrawl/tools/checkstyle/checks/design/OneTopLevelClassCheck.java

```diff
@@ -146,7 +146,8 @@ else if (!isPublic(currentNode)) {
     private static boolean isTypeDef(DetailAST node) {
         return node.getType() == TokenTypes.CLASS_DEF
                 || node.getType() == TokenTypes.ENUM_DEF
-                || node.getType() == TokenTypes.INTERFACE_DEF;
+                || node.getType() == TokenTypes.INTERFACE_DEF
+                || node.getType() == TokenTypes.ANNOTATION_DEF;
```

Comment:
Javadoc was not updated.

Reply:
Fixed.

Annotation types were added to the Google UTs as well.